image/default_image_file.go

```diff
@@ -0,0 +1,123 @@
+package image
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/alibaba/sealer/image/reference"
+	"github.com/alibaba/sealer/image/store"
+	"github.com/alibaba/sealer/utils/compress"
+	"github.com/opencontainers/go-digest"
+
+	"github.com/alibaba/sealer/common"
+	imageutils "github.com/alibaba/sealer/image/utils"
+	v1 "github.com/alibaba/sealer/types/api/v1"
+	"github.com/alibaba/sealer/utils"
+)
+
+type DefaultImageFileService struct {
+	BaseImageManager
+}
+
+func (d DefaultImageFileService) Load(imageSrc string) error {
+	return d.load(imageSrc)
+}
+
+func (d DefaultImageFileService) Save(imageName string, imageTar string) error {
+	return d.save(imageName, imageTar)
+}
+
+func (d DefaultImageFileService) Merge(image *v1.Image) error {
+	panic("implement me")
+}
+
+func (d DefaultImageFileService) save(imageName, imageTar string) error {
+	named, err := reference.ParseToNamed(imageName)
+	if err != nil {
+		return err
+	}
+
+	ima, err := imageutils.GetImage(named.Raw())
+	if err != nil {
+		return err
+	}
+	if imageTar == "" {
+		imageTar = ima.Spec.ID + common.TarGzSuffix
+	}
+	if _, err := os.Stat(imageTar); err == nil {
+		return fmt.Errorf("file %s exist", imageTar)
+	}
+	err = os.MkdirAll(filepath.Dir(imageTar), common.FileMode0644)
+	if err != nil {
+		return fmt.Errorf("failed to create %s,err:%v", imageTar, err)
+	}
+	file, err := os.Create(imageTar)
+	if err != nil {
+		return fmt.Errorf("failed to create %s,err:%v", imageTar, err)
+	}
+	defer file.Close()
+
+	var pathToCompress []string
+	for _, layer := range ima.Spec.Layers {
+		if layer.Hash == "" {
+			continue
+		}
+		layerID := store.LayerID(digest.NewDigestFromEncoded(digest.SHA256, layer.Hash))
+		layerDir := filepath.Join(common.DefaultLayerDir, digest.Digest(layerID).Hex())
+		pathToCompress = append(pathToCompress, layerDir)
+	}
+	tempDir, err := utils.MkTmpdir()
+	if err != nil {
+		return fmt.Errorf("failed to create %s:%v", tempDir, err)
+	}
+	defer utils.CleanDir(tempDir)
+	imageMetadataTempFile := filepath.Join(tempDir, common.DefaultImageMetadataFileName)
+	tempfile, err := os.Create(imageMetadataTempFile)
+	if err != nil {
+		return fmt.Errorf("failed to create image_metadata.yaml,err:%v", err)
+	}
+	imageMetadataFile, err := os.Open(filepath.Join(common.DefaultImageMetaRootDir, ima.Spec.ID+common.YamlSuffix))
+	if err != nil {
+		return fmt.Errorf("failed to read image metadata,err:%v", err)
+	}
+	defer imageMetadataFile.Close()
+	_, err = io.Copy(tempfile, imageMetadataFile)
+	if err != nil {
+		return fmt.Errorf("failed to write image metadata temp file,err:%v", err)
+	}
+	pathToCompress = append(pathToCompress, imageMetadataTempFile)
+	_, err = compress.Compress(file, pathToCompress...)
+	return err
+}
+
+func (d DefaultImageFileService) load(imageSrc string) error {
+	src, err := os.Open(imageSrc)
+	if err != nil {
+		return fmt.Errorf("failed to open %s,%v", imageSrc, err)
```

Comment:
please unify the error format.  "failed to open %s, err: %v"  . Be serious about the format